src/main/java/com/bazel_diff/BazelSourceFileTarget.java

```diff
@@ -17,51 +16,40 @@ interface BazelSourceFileTarget {
 class BazelSourceFileTargetImpl implements BazelSourceFileTarget {
     private String name;
     private byte[] digest;
+    static private ByteBufferPool pool = new ByteBufferPool(1024, 10240); //10kb
 
-    private void digestLargeFile(MessageDigest finalDigest, FileChannel inChannel) throws IOException {
-        int bufferSize = 10240; // 10kb
-        ByteBuffer buffer = ByteBuffer.allocate(bufferSize);
-        while (inChannel.read(buffer) != -1) {
+    private void digest(Hasher finalDigest, InputStream stream) throws Exception {
+        ByteBuffer buffer = pool.borrow();
+        byte[] array = buffer.array(); //Available for non-direct buffers
+        Integer length = 0;
+        while (true) {
+            if (!((length = stream.read(array)) != -1)) break;
             buffer.flip();
-            finalDigest.update(buffer);
+            finalDigest.putBytes(array, 0, length);
             buffer.clear();
         }
+        pool.recycle(buffer);
     }
 
-    private void digestSmallFile(MessageDigest finalDigest, FileChannel inChannel) throws IOException {
-        long fileSize = inChannel.size();
-        ByteBuffer buffer = ByteBuffer.allocate((int) fileSize);
-        inChannel.read(buffer);
-        buffer.flip();
-        finalDigest.update(buffer);
-    }
-
-    BazelSourceFileTargetImpl(String name, byte[] digest, Path workingDirectory, Boolean verbose)
-        throws IOException, NoSuchAlgorithmException {
+    BazelSourceFileTargetImpl(String name, byte[] digest, Path workingDirectory, Boolean verbose) throws Exception {
         this.name = name;
-        MessageDigest finalDigest = MessageDigest.getInstance("SHA-256");
+        Hasher hasher = Hashing.sha256().newHasher();
         if (workingDirectory != null && name.startsWith("//")) {
             String filenameSubstring = name.substring(2);
             String filenamePath = filenameSubstring.replaceFirst(":", "/");
             Path absoluteFilePath = Paths.get(workingDirectory.toString(), filenamePath);
-            try (RandomAccessFile sourceFile = new RandomAccessFile(absoluteFilePath.toString(), "r")) {
-                FileChannel inChannel = sourceFile.getChannel();
-                if (inChannel.size() > 1048576) { // 1mb
-                    digestLargeFile(finalDigest, inChannel);
-                } else {
-                    digestSmallFile(finalDigest, inChannel);
-                }
-                sourceFile.close();
-                inChannel.close();
+
+            try (InputStream stream = new BufferedInputStream(new FileInputStream(absoluteFilePath.toString()))) {
```

Comment:
BufferedInputStream will improve performance a bit compared to RandomAccessFile since we know the reads are sequential